lib/msal-core/src/Utils.ts

```diff
@@ -413,17 +413,34 @@ export class Utils {
         return  !(extraQueryParameters &&  ((extraQueryParameters.indexOf(Constants.login_hint) !== -1 ||  extraQueryParameters.indexOf(Constants.sid) !== -1 )));
     }
 
-     static constructUnifiedCacheExtraQueryParameter(idTokenObject : any, extraQueryParameters ?: string)
-    {
-      if(idTokenObject) {
-          if (idTokenObject.hasOwnProperty(Constants.upn)) {
-              return extraQueryParameters = "&" + Constants.login_hint + "=" + idTokenObject.upn + "&" + Constants.domain_hint + "=" + Constants.organizations;
-          }
-          else  {
-              return extraQueryParameters = "&" + Constants.domain_hint + "=" + Constants.organizations;
-          }
-      }
-        return "";
-    }
+     static constructUnifiedCacheExtraQueryParameter(idTokenObject: any, extraQueryParameters?: string) {
+         if (idTokenObject && idTokenObject.upn) {
+             extraQueryParameters = this.urlRemoveQueryStringParameter(extraQueryParameters, Constants.login_hint);
```

Comment:
I added the code in my branch to check if upn field exists or not but I think it's getting replaced by your code. Can you please keep that logic. 
 if (idTokenObject.hasOwnProperty(Constants.upn)) {